cql3/restrictions/statement_restrictions.cc

```diff
@@ -1394,6 +1395,94 @@ bool is_satisfied_by(
         }, restr);
 }
 
+/// A column's bound, from WHERE restrictions.
+class bound_t {
+    bool _unbounded;
+    bytes_opt _value; // Invalid when _unbounded is true.
+    const abstract_type& _value_type;
+  public:
+    explicit bound_t(const abstract_type& t) : _unbounded(true), _value_type(t) {}
+    bound_t(const abstract_type& t, const bytes_opt& v) : _unbounded(false), _value(v), _value_type(t) {}
+
+    /// True iff *this is a tighter lower bound than \p that.
+    bool is_tighter_lb_than(const bound_t& that) const {
+        if (that._unbounded) {
+            return true; // Anything is tighter than no bound.
+        } else if (this->_unbounded || !this->_value || !that._value) {
```

Comment:
If !this->_value, and if that->_value, then isn't that a tighter bound?

Reply:
Good point; fixed.